### ISSUE TITLE
Include searching a username with a tag functionality

### ISSUE THREAD

**Issue by Gregory-rt** [sherlock-project/sherlock#2164]
### Description

With many sites (League of Legends, Valorant, Discord in the past) having introduced a username + tag way of identifying an account (e.g Theo#1213), I would like to propose the accomodation of such sites by allowing the user to input an optional tag when searching with sherlock. This would allow the addition of such sites to sherlock, since adding them without a way to input the tag would never result in a positive response from the site. 

My proposal would be to add a --tag optional argument, which would be used when searching through sites that have been identified as tag_required in the json file. These sites could also have a # in the url of the json file indicating where the tag should be placed. (e.g https://www.leagueofgraphs.com/summoner/vn/Sophie-1911 this would be "url": "https://www.leagueofgraphs.com/summoner/vn/{}-#" since the username of the example is Sophie and 1911 is the tag).

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by roxerj** [comment]
Working on it, but using a "#" as a placeholder for a name tag is probably not a good idea since some URLs might already have this symbol. I think it's better to use "<>" or a second "{}". 
I'm trying to add a new property when adding a website, something like a boolean variable "isTagRequired", and if set to True, user would be prompted to add a tag after the username, for example "sherlock user123#1337"

**Comment by Canusmellthatbag** [comment]
is the Sherlock project also capable or searching by email and or phone
number to find accounts?; inquiring minds would love to know

**Comment by Gregory-rt** [comment]
> Working on it, but using a "#" as a placeholder for a name tag is probably not a good idea since some URLs might already have this symbol. I think it's better to use "<>" or a second "{}". I'm trying to add a new property when adding a website, something like a boolean variable "isTagRequired", and if set to True, user would be prompted to add a tag after the username, for example "sherlock user123#1337"

I have been working on it as well. Your recommendation about # makes sense to me, I proposed it since it would be intuitive and would make the code more readable.  Asking the user to add a tag would only make sense to me if he was using the --site optional argument which specifies which site to search for.

**Comment by roxerj** [comment]
You're right, so I see 2 options:
1) Only search for a username with tag in websites where isTagRequired is set to True; e.g. "sherlock user123#1337 --tag" would only check websites with isTagRequired. Otherwise "sherlock user123#1337" would work the same way it's been working till now.
2) Dynamically check for a "#" symbol in the username arg, so if you enter "sherlock user123#1337", the tag would be detected automatically without using the "--tag" flag. Then proceed to only check the websites with isTagRequired.
> Asking the user to add a tag would only make sense to me if he was using the --site optional argument which specifies which site to search for.

**Comment by Gregory-rt** [comment]
I have already done something pretty similar to the first one (only difference is that the user would have to type "sherlock user123 --tag 1337" for the input you mentioned) and the same functionality (ignoring all sites were tag isnt required). I was waiting for a member of sherlock to respond before opening a pr for that solution. Would you mind working on the second proposal you gave, so that the team could choose between the two?

**Comment by roxerj** [comment]
Yeah, sure, not a problem, even if your solution gets accepted first, I don't mind. Still having fun here

**Comment by ppfeister** [comment]
Note that `--tag`/`--tags` is also being discussed for use as in #2088

**Comment by roxerj** [comment]
I have already created a PR, #2169
